tests/actor/test_runner.py

```diff
@@ -0,0 +1,71 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Simple tests for the actor runner."""
+
+import asyncio
+
+from frequenz.sdk.actor import actor, run
+
+
+@actor
+class SleepyActor:
+    """A test actor that sleeps a short time."""
+
+    def __init__(self, name: str) -> None:
+        """Initialize the test sleepy actor.
+
+        Args:
+            name: the name of the test sleepy actor.
+        """
+        self.name = name
+        self.is_joined = False
+
+    async def run(self) -> None:
+        """Run the test sleepy actor."""
+        await asyncio.sleep(0.1)
+        self.is_joined = True
+        # pylint: disable=protected-access,no-member
+        await self._stop()  # type: ignore[attr-defined]
+
+
+@actor
+class FaultyActor:
+    """A test faulty actor."""
+
+    def __init__(self, name: str) -> None:
+        """Initialize the test faulty actor.
+
+        Args:
+            name: the name of the test sleepy actor.
+        """
+        self.name = name
+        self.is_cancelled = False
+
+    async def run(self) -> None:
+        """Run the test faulty actor.
+
+        Raises:
+            CancelledError: after sleeping a short time.
+        """
+        await asyncio.sleep(0.1)
+        self.is_cancelled = True
+        raise asyncio.CancelledError("Faulty Actor 1 failed")
+
+
+async def test_all_actors_done() -> None:
+    """Test the completion of all actors."""
+
+    sleepy_actors = [SleepyActor(f"sleepy_actor_{i}") for i in range(3)]
+    await run(*sleepy_actors)
+    for sleepy_actor in sleepy_actors:
+        assert sleepy_actor.is_joined, f"{sleepy_actor.name} was not joined"
```

Comment:
You probably want to add a timeout here, otherwise if `run()` is misbehaving and waiting forever, tests will also wait forever. Same for the other test.

You should probably also check that the time elapsed in `run()` is more than `0.1` to make sure it also waited. Also, when you start playing with time, I strongly recommend using async-solipsism (makes the time pass in async code) and timemachine (fake the tiem), so tests don't really take that much time and we avoid flaky tests. 

Examples:

* https://github.com/frequenz-floss/frequenz-sdk-python/blob/v0.x.x/tests/timeseries/test_resampling.py
* https://github.com/frequenz-floss/frequenz-sdk-python/blob/v0.x.x/tests/actor/test_resampling.py

https://docs.python.org/3/library/asyncio-task.html#timeouts (warning: many of the timeout functions are only available in Python 3.11+).

Reply:
ok, I'll amend the tests following the suggestions and examples

Reply:
Updated based on the suggestion and examples. Please have another look at the test to check if that's pretty much what you have in mind